avatars/microphone-worklet.js

```diff
@@ -1,63 +1,62 @@
-const numTicks = 1;
-
-let tick = 0;
-let sampleSum = 0;
-let numSamples = 0;
-let muted = true;
-let emitVolume = false;
-let emitBuffer = false;
-
-const queue = [];
-let queueLength = 0;
-const maxQueueLength = 4000;
 class VolumeProcessor extends AudioWorkletProcessor {
   constructor() {
     super();
 
+    this.numTicks = 1;
+    this.tick = 0;
+    this.sampleSum = 0;
+    this.muted = true;
+    this.emitVolume = false;
+    this.emitBuffer = false;
+    this.queue = [];
+    this.queueLength = 0;
+    this.maxQueueLength = 4000;
+
     this.port.addEventListener('message', e => {
       const data = JSON.parse(e.data);
       const {method} = data;
       if (method === 'options') {
         const {args} = data;
         if (args.muted !== undefined) {
-          muted = args.muted;
+          this.muted = args.muted;
         }
         if (args.emitVolume !== undefined) {
-          emitVolume = args.emitVolume;
+          this.emitVolume = args.emitVolume;
         }
         if (args.emitBuffer !== undefined) {
-          emitBuffer = args.emitBuffer;
+          this.emitBuffer = args.emitBuffer;
         }
       }
     });
     this.port.start();
   }
   process(inputs, outputs) {
+    try {

```

Comment:
Is this try catch needed? If there is a bug, it should be fixed. If there is no bug then we can let it crash.